### PR TITLE
Quality of life readability improvements

### PR DESCRIPTION
**Pull request by fzakaria**
Series of changes that merely change some function names and introduce a header file to help make the code more legible.

- introduce patchelf.h header file
- rename findSection to findSectionHeader to better distinguish sections from section-headers
- rename findSection2 to tryFindSection to better distinguish it's optional
- rename findSection3 to findSectionIndex

### CODE REVIEW

File: src/patchelf.cc [hunk 1]
```diff
@@ -41,6 +41,7 @@
 #include <unistd.h>
 
 #include "elf.h"
+#include "patchelf.h"
```
Comment (Mic92) [inline src/patchelf.cc:44]: Not quite sure why patchelf.h is not found here.

**Comment by fzakaria** [comment]
Hmm.
I see it adding the current directory to the **g++** call
```
make[1]: Entering directory '/gha/patchelf-0.14.3/src'
g++ -DPACKAGE_NAME=\"patchelf\" -DPACKAGE_TARNAME=\"patchelf\" -DPACKAGE_VERSION=\"0.14.3\" -DPACKAGE_STRING=\"patchelf\ 0.14.3\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"patchelf\" -DVERSION=\"0.14.3\" -I.    -Wall -std=c++17 -D_FILE_OFFSET_BITS=64     -D_FORTIFY_SOURCE=2 -fstack-protector-strong -Wformat -Werror=format-security -O2 -static -MT patchelf.o -MD -MP -MF .deps/patchelf.Tpo -c -o patchelf.o patchelf.cc
```

You can see `-I.`

**Comment by fzakaria** [comment]
@Mic92 i started alpine directly and entered the nix-shell and it builds.
```
/app # cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.15.0
PRETTY_NAME="Alpine Linux v3.15"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://bugs.alpinelinux.org/"

/app # ls src/patchelf.h 
src/patchelf.h

/app # make
Making all in src
make[1]: Entering directory '/app/src'
g++ -DPACKAGE_NAME=\"patchelf\" -DPACKAGE_TARNAME=\"patchelf\" -DPACKAGE_VERSION=\"0.14.3\" -DPACKAGE_STRING=\"patchelf\ 0.14.3\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"patchelf\" -DVERSION=\"0.14.3\" -I.    -Wall -std=c++17 -D_FILE_OFFSET_BITS=64     -g -O2 -MT patchelf.o -MD -MP -MF .deps/patchelf.Tpo -c -o patchelf.o patchelf.cc
mv -f .deps/patchelf.Tpo .deps/patchelf.Po
g++ -Wall -std=c++17 -D_FILE_OFFSET_BITS=64     -g -O2   -o patchelf patchelf.o  
make[1]: Leaving directory '/app/src'
Making all in tests
```

Not sure the pull-request action to download the distribution. Could it be not including **patchelf.h** somehow ?

File: src/patchelf.cc (hunk 1 above)
Comment (fzakaria) [reply src/patchelf.cc:44]: Fixed it! It was missing from the **automake** dist file.

File: src/Makefile.am [hunk 2]
```diff
@@ -19,4 +19,4 @@ endif
 
 bin_PROGRAMS = patchelf
 
-patchelf_SOURCES = patchelf.cc elf.h
+patchelf_SOURCES = patchelf.cc elf.h patchelf.h
```
Comment (fzakaria) [inline src/Makefile.am:22]: @Mic92 found the reason!
I am not very familiar with autotools, so this is all new to me :)

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**